### ISSUE TITLE
DS data page cancel button resets form

### ISSUE THREAD

**Issue by AnnaGingle** [cisagov/manage.get.gov#2700]
### Issue description

The DS data form cancel button resets the form, rather than reverting to the previous read-only state. This isn't consistent with other `Cancel` button behavior. There are also accessibility issues on this page that may need to be addressed through design.

### Acceptance Criteria

- [x] Update the design so the cancel button behavior is consistent with our existing patterns.
- [x] Audit page for accessibility. [Audit here](https://docs.google.com/spreadsheets/d/1XHN2fBZMNGa2-zpFOo8m6d2b0tEhOgjCrflVDbZS1jQ/edit?gid=835883067#gid=835883067).
- [x] Approval from product
- [x] Coordinate with dev to create tickets

### Additional Context

See [slack thread in private design review channel](https://cisa-corp.slack.com/archives/C07HJTNV066/p1724771386504799).

https://github.com/user-attachments/assets/9a923318-110f-4d41-9488-68580107a1cb

### Issue Links
- DS data page: Associate delete button with according DS record #2773

**Comment by gabydisarli** [comment]
@katypies I've updated the link, can you see it now?

**Comment by AnnaGingle** [comment]
Documenting issues here: https://docs.google.com/spreadsheets/d/1XHN2fBZMNGa2-zpFOo8m6d2b0tEhOgjCrflVDbZS1jQ/edit?gid=835883067#gid=835883067

**Comment by AnnaGingle** [comment]
The [designs](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-registar?node-id=9244-22349&t=JkMWw7KF0iFTAmDT-4) are ready for review
